src/index.mjs

```diff
@@ -65,16 +65,28 @@ export function listen(options) {
 
   const timeoutFn = options.timeoutFn || requestIdleCallback;
 
+  // ray test touch <
+  const prefetchChunks = options.prefetchChunks;
+  // ray test touch >
+
+  // ray test touch <
+  const prefetchHandler = urls => {
+    prefetch(urls, options.priority).then(isDone).catch(err => {
+      isDone(); if (options.onError) options.onError(err);
+    });
+  };
+  // ray test touch >
+
   const observer = new IntersectionObserver(entries => {
     entries.forEach(entry => {
       if (entry.isIntersecting) {
         observer.unobserve(entry = entry.target);
         // Do not prefetch if will match/exceed limit
         if (toPrefetch.size < limit) {
           toAdd(() => {
-            prefetch(entry.href, options.priority).then(isDone).catch(err => {
-              isDone(); if (options.onError) options.onError(err);
-            });
+            // ray test touch <
+            prefetchChunks ? prefetchChunks(entry, prefetchHandler) : prefetchHandler(entry.href);
```

Comment:
I think once we have explained why the `prefetchChunks` option is needed in this PR, it would be good to also write out some docs additions to the project README so that we have API docs for this change too.

Reply:
I added some comment above regarding this.
Once it's confirmed, I'm going to update README and PR comment accordingly.

Reply:
Done